mithril-common/src/entities/signed_entity_type.rs

```diff
@@ -1,46 +1,64 @@
-use strum::IntoEnumIterator;
-use strum_macros::{Display, EnumIter, EnumString, FromRepr};
+use strum_macros::Display;
+
+use crate::StdError;
+
+use super::{Beacon, Epoch};
+
+/// Database representation of the SignedEntityType::MithrilStakeDistribution value
+pub const ENTITY_TYPE_MITHRIL_STAKE_DISTRIBUTION: usize = 0;
+
+/// Database representation of the SignedEntityType::CardanoStakeDistribution value
+pub const ENTITY_TYPE_CARDANO_STAKE_DISTRIBUTION: usize = 1;
+
+/// Database representation of the SignedEntityType::CardanoImmutableFilesFull value
+pub const ENTITY_TYPE_CARDANO_IMMUTABLE_FILES_FULL: usize = 2;
 
 /// The signed entity type that represents a type of data signed by the Mithril
 /// protocol Note: Each variant of this enum must be associated to an entry in
 /// the `signed_entity_type` table of the signer/aggregator nodes. The variant
 /// are identified by their discriminant (i.e. index in the enum), thus the
 /// modification of this type should only ever consist of appending new
 /// variants.
-#[derive(Display, FromRepr, EnumString, EnumIter, Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Display, Debug, Clone, PartialEq, Eq)]
 #[strum(serialize_all = "PascalCase")]
 pub enum SignedEntityType {
```

Comment:
Maybe `SignedEntityType` could be renamed as `SignedEntityTypeWithBeacon` and we could add this strum directives:
- `#[strum_discriminants(name(SignedEntityType))]`
- `#[strum_discriminants(derive(EnumIter))]` (to allow listing the discriminant variants)